### PR TITLE
WINC-699: [build] Upgrade to Go 1.17

### PR DESCRIPTION
**Pull request by saifshaikh48**
This PR is to build WMCO with golang 1.17, in line with the OCP bump
https://github.com/openshift/ocp-build-data/pull/1143.

Note that we will have issues [adding new resources](https://github.com/operator-framework/operator-sdk/issues/5284) via operator-sdk until the [SDK moves to go 1.17](https://github.com/operator-framework/operator-sdk/issues/5208) as well.

### CODE REVIEW

File: .ci-operator.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@
 build_root_image:
```
Comment (jrvaldes) [inline .ci-operator.yaml:2]: Do we also need to update `go.mod` file? i.e.
```
go mod edit -go=1.17
```

Comment (saifshaikh48) [reply .ci-operator.yaml:2]: Yes, good catch

**Review by selansen — APPROVED** [review]
/lgtm

**Review by sebsoto — APPROVED** [review]

**Pull request merged**